src/System.Resources.Extensions/src/System/Resources/Extensions/PreserializedResourceWriter.cs

```diff
@@ -25,14 +33,43 @@ partial class PreserializedResourceWriter
             RuntimeResourceSetFullyQualifiedName :
             ResSetTypeName;
 
+        // a collection of primitive types in a dictionary, indexed by type name
+        // using a comparer which handles type name comparisons similar to what
+        // is done by refelection
+        private static readonly Dictionary<string, Type> s_primitiveTypes = new[]
+        {
+            typeof(string),
+            typeof(int),
+            typeof(bool),
+            typeof(char),
+            typeof(byte),
+            typeof(sbyte),
+            typeof(short),
+            typeof(long),
+            typeof(ushort),
+            typeof(uint),
+            typeof(ulong),
+            typeof(float),
+            typeof(double),
+            typeof(decimal),
+            typeof(DateTime),
+            typeof(TimeSpan)
+
+            // The following primitive types do not define a conversion from string
+            // typeof(byte[]),
+            // typeof(Stream)
+        }.ToDictionary(t => t.FullName, TypeNameComparer.Instance);
```

Comment:
Nit: you could avoid the startup cost of LINQ by instead doing something like:
```C#
private static readonly Dictionary<string, Type> s_primitiveTypes = new Dictionary<string, Type>(16, TypeNameComparer.Instance)
{
    { typeof(string).Name, typeof(string) },
    { typeof(int).Name, typeof(int) },
    { typeof(bool).Name, typeof(bool) },
    ...
};
```

Reply:
Will do.  I was annoyed by all this duplication but I only need to deal with it once.